test/e2e/framework.go

```diff
@@ -1577,7 +1578,39 @@ func (data *TestData) PodWaitFor(timeout time.Duration, name, namespace string,
 // until the provided timeout expires).
 func (data *TestData) podWaitForRunning(timeout time.Duration, name, namespace string) error {
```

Comment:
it makes no sense IMO to modify this function as such. The purpose of the function is pretty clear, and it's reflected in the function comment:

> podWaitForRunning polls the k8s apiserver until the specified Pod is in the "running" state

you should define a new function for your use case, and have a helpful comment describing what it does

Reply:
Added podWaitForAvailable function.